src/Eloquent/DocumentModel.php

```diff
@@ -113,18 +118,6 @@ public function fromDateTime($value)
         return new UTCDateTime($value);
     }
 
-    /** @inheritdoc */
-    protected function asDateTime($value)
-    {
-        // Convert UTCDateTime instances to Carbon.
-        if ($value instanceof UTCDateTime) {
-            return Date::instance($value->toDateTime())
-                ->setTimezone(new DateTimeZone(date_default_timezone_get()));
-        }
-
-        return parent::asDateTime($value);
-    }
```

Comment:
This is not necessary at model level because the conversion is done in query result.